storage/response_cache.go

```diff
@@ -90,14 +90,13 @@ func (rc *ResponseCache) GetResponse(cmdID proto.ClientCmdID, reply proto.Respon
 	// If the response is in the cache or we experienced an error, return.
 	rwResp := proto.ReadWriteCmdResponse{}
 	key := keys.ResponseCacheKey(rc.raftID, &cmdID)
-	if ok, err := engine.MVCCGetProto(rc.engine, key, proto.ZeroTimestamp, true, nil, &rwResp); ok || err != nil {
-		if err == nil && rwResp.GetValue() != nil {
-			gogoproto.Merge(reply.(gogoproto.Message), rwResp.GetValue().(gogoproto.Message))
+	ok, err := engine.MVCCGetProto(rc.engine, key, proto.ZeroTimestamp, true, nil, &rwResp)
+	if ok && err == nil {
+		if respProto, isMsg := rwResp.GetValue().(gogoproto.Message); isMsg && respProto != nil {
```

Comment:
The `respProto != nil` check is unnecessary because A) `gogoproto.Message` is an interface B) `gogoproto.Merge` is safe when its second argument is nil.

There should be an `else` case to handle the possibility that we get a non-`gogoproto.Message` instead of just silently ignoring it (which can happen if none of the options in the `onlyone` are set, or if we add any primitive types to the `onlyone`). The old code had the implicit `else panic()` of an unchecked type assertion.